src/main/java/randoop/generation/WeightedComponentManager.java

```diff
@@ -0,0 +1,53 @@
+package randoop.generation;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import randoop.main.GenInputsAbstract;
+import randoop.sequence.Sequence;
+
+/**
+ * WeightedComponentManager extends the functionality of ComponentManager by adding needed
```

Comment:
"needed" is not very descriptive here.  Could you document what this class does?  I think you can do so in just a sentence or two, and that will make this code stand on its own.